src/Data/Profunctor/Choice.hs

```diff
@@ -167,7 +178,7 @@ instance (Choice p, Choice q) => Choice (Sum p q) where
   right' (R2 q) = R2 (right' q)
   {-# INLINE right' #-}
 
-instance (Functor f, Choice p) => Choice (Tannen f p) where
+instance (Functor f, Bifunctor p, Choice p) => Choice (Tannen f p) where
```

Comment:
Same here - this `Bifunctor` constraint is not right

Reply:
Unfortunately, the `Bifunctor` constraint here is forced. See my coments above about `Cayley`, which is a close cousin of `Tannen`.